### PR TITLE
feature(cb2-6070): archive current tech record

### PR DESCRIPTION
**Pull request by tahayazici**
## Archive Current Tech Record

- Introduce an archiving workflow.
- Reuse the promotion component.
- Add a service method to handle this.

[CB2-6070](https://dvsa.atlassian.net/browse/CB2-6070)

### CODE REVIEW

File: src/app/features/tech-record/components/tech-record-title/tech-record-title.component.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  @Input() recordActions: TechRecordActions = TechRecordActions.NONE;
+
+  queryableRecordActions: string[] = [];
+  currentTechRecord$!: Observable<TechRecordModel | undefined>;
+
+  constructor(private technicalRecordService: TechnicalRecordService) { }
+
+  ngOnInit(): void {
+    this.queryableRecordActions = this.recordActions.split(',');
+    console.log(this.queryableRecordActions)
```
Comment (mcadenas-bjss) [inline src/app/features/tech-record/components/tech-record-title/tech-record-title.component.ts:22]:
```suggestion
```

Comment (tahayazici) [reply src/app/features/tech-record/components/tech-record-title/tech-record-title.component.ts]: Removed.

File: src/app/features/tech-record/components/tech-promote/tech-promote.component.ts [hunk 2]
```diff
@@ -1,58 +1,80 @@
...
     private errorService: GlobalErrorService,
     private route: ActivatedRoute,
     private router: Router,
     private routerStore: Store<RouterReducerState>,
     private techRecordsStore: Store<TechnicalRecordServiceState>,
     private technicalRecordService: TechnicalRecordService
   ) {
-    this.errorService.clearErrors();
-
     this.vehicleTechRecord$ = this.technicalRecordService.selectedVehicleTechRecord$.pipe(
```
Comment (gjulien-bjss) [inline src/app/features/tech-record/components/tech-promote/tech-promote.component.ts:44]: It seems this is basically just `this.technicalRecordService.techRecord$`

File: src/app/features/tech-record/components/tech-promote/tech-promote.component.ts [hunk 3]
```diff
@@ -1,58 +1,80 @@
...
-      tap(
-        vehicleTechRecord =>
-          (this.techRecord = cloneDeep(vehicleTechRecord?.techRecord.find(record => record.statusCode === StatusCodes.PROVISIONAL))!)
-      )
+      tap(vehicle => this.technicalRecordService.viewableTechRecord$(vehicle!).subscribe(techRecord => (this.techRecord = techRecord)))
```
Comment (mcadenas-bjss) [inline src/app/features/tech-record/components/tech-promote/tech-promote.component.ts:45]: A subscribe inside a tap feel counter intuitive. Would it not be nicer to pipe the emitted value of the outer observable into a new inner one? try using `switchMap` like `...pipe(<take>, switchMap( v => this.technicalRecordService.viewableTechRecord$(v!))).subscribe...`

**Review by naathanbrown — DISMISSED** [review]

File: src/app/features/tech-record/components/tech-promote/tech-promote.component.ts (hunk 2 above)
Comment (tahayazici) [reply src/app/features/tech-record/components/tech-promote/tech-promote.component.ts:44]: Using that instead now, thanks.

File: src/app/features/tech-record/components/tech-promote/tech-promote.component.ts (hunk 3 above)
Comment (tahayazici) [reply src/app/features/tech-record/components/tech-promote/tech-promote.component.ts:45]: Addressing your comment above resolved this issue as well.

**Review by mcadenas-bjss — APPROVED** [review]

**Review by naathanbrown — APPROVED** [review]

**Review by naathanbrown — APPROVED** [review]

**Review by gjulien-bjss — APPROVED** [review]

**Pull request merged**